src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -0,0 +1,118 @@
+package frc.robot.subsystems;
+
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.ctre.phoenix.motorcontrol.can.WPI_VictorSPX;
+import com.ctre.phoenix6.configs.Pigeon2Configuration;
+import com.ctre.phoenix6.hardware.Pigeon2;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.RobotMap;
+
+public class DriveSubsystem extends SubsystemBase {
+    private final WPI_TalonSRX leftFrontMotor;
+    private final WPI_VictorSPX rightFrontMotor;
+    private final WPI_VictorSPX leftBackMotor;
+    private final WPI_TalonSRX rightBackMotor;
+    private final Field2d field2d;
+    private final DifferentialDriveOdometry differentialDriveOdometry;
+    private Pigeon2 pigeon2;
+
+    public DriveSubsystem() {
+        leftBackMotor = new WPI_VictorSPX(RobotMap.DRIVE_LEFT_BACK_MOTOR_ID);
+        rightBackMotor = new WPI_TalonSRX(RobotMap.DRIVE_RIGHT_BACK_MOTOR_ID);
+        leftFrontMotor = new WPI_TalonSRX(RobotMap.DRIVE_LEFT_FRONT_MOTOR_ID);
+        rightFrontMotor = new WPI_VictorSPX(RobotMap.DRIVE_RIGHT_FRONT_MOTOR_ID);
+        pigeon2 = new Pigeon2(RobotMap.PIGEON_ID);
+        leftBackMotor.configFactoryDefault();
+        rightBackMotor.configFactoryDefault();
+        leftFrontMotor.configFactoryDefault();
+        rightFrontMotor.configFactoryDefault();
+        pigeon2.getConfigurator().apply(new Pigeon2Configuration());
+
+        leftFrontMotor.setInverted(true);
+        leftBackMotor.setInverted(true);
+        leftFrontMotor.setSensorPhase(true);
+
+        this.field2d = new Field2d();
+        SmartDashboard.putData("field2d" ,field2d);
+
+        differentialDriveOdometry = new DifferentialDriveOdometry(
+                new Rotation2d(getAngleDegrees()),
```

Comment:
this is a wrong use of `Rotation2d` please see the way you used it in `updateOdometery`